discovery/gossiper.go

```diff
@@ -2514,15 +2521,22 @@ func (d *AuthenticatedGossiper) updateChannel(info *channeldb.ChannelEdgeInfo,
 	if timestamp <= edge.LastUpdate.Unix() {
 		timestamp = edge.LastUpdate.Unix() + 1
 	}
+
+	// Check whether optional fields are present.
+	msgFlags := 1
+	if edge.MaxHTLC == 0 {
```

Comment:
It looks like we will always set `MaxHTLC` before calling this, perhaps we don't need to have this conditional, and we can always set have the field enabled? Perhaps I'm missing something tho

Reply:
I think we can keep this and remove the setting before calling `updateChannel`.